src/OpenTelemetry/Metrics/MetricStreamIdentity.cs

```diff
@@ -37,9 +37,30 @@ public MetricStreamIdentity(Instrument instrument, MetricStreamConfiguration met
             this.HistogramBucketBounds = (metricStreamConfiguration as ExplicitBucketHistogramConfiguration)?.CopiedBoundaries;
             this.HistogramRecordMinMax = (metricStreamConfiguration as HistogramConfiguration)?.RecordMinMax ?? true;
 
+#if NETSTANDARD2_1 || NET6_0_OR_GREATER
+            HashCode hashCode = default;
+            hashCode.Add(this.InstrumentType.GetHashCode());
+            hashCode.Add(this.MeterName.GetHashCode());
+            hashCode.Add(this.MeterVersion.GetHashCode());
+            hashCode.Add(this.InstrumentName.GetHashCode());
+            hashCode.Add(this.HistogramRecordMinMax.GetHashCode());
+            hashCode.Add(this.Unit.GetHashCode());
+            hashCode.Add(this.Description.GetHashCode());
+            hashCode.Add(this.ViewId?.GetHashCode() ?? 0);
```

Comment:
I was just looking at the [source to HashCode.Add](https://github.com/dotnet/runtime/blob/dcda1e08cb8f421877388be2d685c009a31b1845/src/libraries/System.Private.CoreLib/src/System/HashCode.cs#L304) it seems to do the coalescing automatically so you could probably simplify this stuff to just pass the raw value eg:

```csharp
hashCode.Add(this.ViewId);
```